### PR TITLE
Consume vmm-sys-util crate from rust-vmm in Firecracker

### PR DESCRIPTION
**Pull request by iulianbarbu**
In this commit `net_util`, `sys_util` & `fc_util` crates were combined
into a single crate named `utils`. Also, `utils` crate is a wrapper
over vmm-sys-util, which is now consumed in Firecracker, replacing `sys_util`.

Signed-off-by: Iulian Barbu <iul@amazon.com>

## Reason for This PR

#1346 

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria. Where there are two options, keep one.]`  
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] Either this PR is linked to an issue, or, the reason for this PR is
      clearly provided. 
- [x] The description of changes is clear and encompassing.
- [x] Either no docs need to be updated as part of this PR, or, the required
      doc changes are included in this PR. Docs in scope are all `*.md` files
      located either in the repository root, or in the `docs/` directory.
- [x] Either no code has been touched, or, code-level documentation for touched
      code is included in this PR.
- [x] Either no API changes are included in this PR, or, the API changes are
      reflected in `firecracker/swagger.yaml`.
- [x] Either the changes in this PR have no user impact, or, the changes in
      this PR have user impact and have been added to the `CHANGELOG.md` file.
- [x] Either no new `unsafe` code has been added, or, the newly added `unsafe`
      code is unavoidable and properly documented.

### CODE REVIEW

**Comment by acatangiu** [comment]
@firecracker-microvm/compute-capsule I'd like to prioritize this PR so that we unblock https://github.com/firecracker-microvm/firecracker/issues/1386

File: utils/src/time.rs [hunk 1]
```diff
@@ -224,4 +224,7 @@ mod tests {
 
         assert!(seconds_to_nanoseconds(9_223_372_037).is_none());
     }
+
+    #[test]
+    fn test_stdin() {}
```
Comment (andreeaflorescu) [inline utils/src/time.rs]: What's up with this one?

File: utils/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#[macro_use]
+extern crate vmm_sys_util;
+use std::os::raw::c_int;
+pub use vmm_sys_util::{errno, eventfd, fam, ioctl, signal, tempdir, tempfile, terminal};
+
+/// Wrapper to interpret syscall exit codes and provide a rustacean `io::Result`
```
Comment (andreeaflorescu) [inline utils/src/lib.rs]: Any reason to keep this functionality in the root of the module?

File: utils/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#[macro_use]
+extern crate vmm_sys_util;
+use std::os::raw::c_int;
+pub use vmm_sys_util::{errno, eventfd, fam, ioctl, signal, tempdir, tempfile, terminal};
```
Comment (andreeaflorescu) [inline utils/src/lib.rs]: It looks like errno, fam, ioctl, tempfile and tempdir are not yet used in the Firecracker code. Should we re-export them in this case?

File: utils/src/time.rs (hunk 1 above)
Comment (iulianbarbu) [reply utils/src/time.rs]: Upsy, just a silly leftover. Removed it. Take a look here:  [
1568750](https://github.com/firecracker-microvm/firecracker/pull/1413/commits/1568750cd716861296faee66063ec64f9cb722a3).

File: utils/src/lib.rs (hunk 3 above)
Comment (iulianbarbu) [reply utils/src/lib.rs]: You're right, we can remove fam, ioctl, tempfile and tempdir, but errno is still used here: [019c3e7](https://github.com/firecracker-microvm/firecracker/pull/1413/commits/019c3e795f9d43f52598e12fe16f455ee20f6a1c)

File: utils/src/lib.rs (hunk 2 above)
Comment (iulianbarbu) [reply utils/src/lib.rs]: This is exposed in (sys_util/srs/lib.rs)[https://github.com/firecracker-microvm/firecracker/blob/master/sys_util/src/lib.rs#L28] right now. As a consequence, I add it also into `utils/src/lib.rs`.

**Comment by acatangiu** [comment]
Blimey! 64 files changed to re-review

**Comment by iulianbarbu** [comment]
> Blimey! 64 files changed to re-review

Is the same thing, basically, but spread out on multiple commits. :D It should be faster to re-review.

File: utils/src/lib.rs (hunk 2 above)
Comment (andreeaflorescu) [reply utils/src/lib.rs]: Can we add an issue to move this?

Comment (iulianbarbu) [reply utils/src/lib.rs]: An issue to move it in a specific place, or an issue opened for discussions?
For this PR, is it OK for you if I let this functionality here?

Comment (andreeaflorescu) [reply utils/src/lib.rs]: We should create an issue that basically says move the floating code from utils/lib.rs that treats syscall exit codes to a separate module. It's okay to leave the functionality as is in this PR.

Comment (iulianbarbu) [reply utils/src/lib.rs]: Sure, thanks. I will open an issue.

**Pull request merged**